### PR TITLE
8333582: Update CLDR to Version 46.0

### PR DESCRIPTION
**Pull request by naotoj**
Upgrading the CLDR to version 46.0. A corresponding CSR has also been drafted.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8333584](https://bugs.openjdk.org/browse/JDK-8333584) to be approved

### Issues
 * [JDK-8333582](https://bugs.openjdk.org/browse/JDK-8333582): Update CLDR to Version 46.0 (**Enhancement** - P3)
 * [JDK-8333584](https://bugs.openjdk.org/browse/JDK-8333584): Update CLDR to Version 46.0 (**CSR**)

### Reviewers
 * [Steven Loomis](https://openjdk.org/census#srl) (@srl295 - Committer) Review applies to [665bc11e](https://git.openjdk.org/jdk/pull/21772/files/665bc11e979e2b3dd6810204f2019db210e3ce27)
 * [Justin Lu](https://openjdk.org/census#jlu) (@justin-curtis-lu - Committer) Review applies to [4da007e7](https://git.openjdk.org/jdk/pull/21772/files/4da007e720d5d382aa953d4f2cef6d368d9cdb3c)
 * [Joe Wang](https://openjdk.org/census#joehw) (@JoeWang-Java - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21772/head:pull/21772` \
`$ git checkout pull/21772`

Update a local copy of the PR: \
`$ git checkout pull/21772` \
`$ git pull https://git.openjdk.org/jdk.git pull/21772/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21772`

View PR using the GUI difftool: \
`$ git pr show -t 21772`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21772.diff">https://git.openjdk.org/jdk/pull/21772.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21772#issuecomment-2448481930)
</details>

### CODE REVIEW

**Review by srl295 — APPROVED** [review]
Spot checked the data and tests. Checked the provider changes, all LGTM.

**Review by justin-curtis-lu — APPROVED** [review]
Data changes look to be CLDR 46. Went through the tests as well.
Change in build to add the aliases of the preferred value for a deprecated zone LGTM.

Left minor comments.

**Comment by naotoj** [comment]
Thanks, Justin

> Left minor comments.

They are both leftovers. Removed.

File: test/jdk/sun/text/resources/LocaleDataTest.java [hunk 1]
```diff
@@ -41,7 +41,7 @@
  *      8187946 8195478 8181157 8179071 8193552 8202026 8204269 8202537 8208746
  *      8209775 8221432 8227127 8230284 8231273 8233579 8234288 8250665 8255086
  *      8251317 8274658 8283277 8283805 8265315 8287868 8295564 8284840 8296715
- *      8301206 8303472 8317979 8306116 8174269
+ *      8301206 8303472 8317979 8306116 8174269 8333582
```
Comment (JoeWang-Java) [inline test/jdk/sun/text/resources/LocaleDataTest.java:44]: Similar situation as TimeZoneNameTest.java? Looks like no other change than the bug id.

**Review by JoeWang-Java — APPROVED** [review]
Looks good to me.

File: test/jdk/sun/text/resources/LocaleDataTest.java (hunk 1 above)
Comment (naotoj) [reply test/jdk/sun/text/resources/LocaleDataTest.java:44]: There are changes in `LocaleData.cldr` data file, which is an input to this test case. Everytime the data file is modified, the bug id is recorded here.

**Pull request closed without merging**